react/week3/todo-week3/src/todo-app.js

```diff
@@ -0,0 +1,210 @@
+import { useState, useEffect } from "react"
+
+const URL =
+  "https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw"
+
+function Todo({ todo, Check, Delete, Edit, editTime }) {
+  const [Editing, setEditing] = useState(false)
+
+  const { description, checked, deadline, id } = todo
+
+  function editing(event) {
+    const value = event.target.value
+    Edit(id, value)
+  }
+
+  function onEditTime(event) {
+    const value = event.target.value
+    editTime(id, value)
+  }
+
+  return (
+    <div>
+      <div style={{ textDecoration: checked ? "line-through" : "none" }}>
+        {Editing ? (
+          <input type="text" value={description} onChange={editing} />
+        ) : (
+          <h3>{description}</h3>
+        )}
+        {Editing ? (
+          <input
+            type="date"
+            id="deadline"
+            name="deadline"
+            value={deadline}
+            onChange={onEditTime}
+          />
+        ) : (
+          <h4>{deadline}</h4>
+        )}
+      </div>
+      <input type="checkbox" checked={checked} onChange={Check} />
+      <button onClick={Delete}>Delete</button>
+      <button onClick={() => setEditing(!Editing)}>
+        {Editing ? "Update" : "Edit"}
+      </button>
+    </div>
+  )
+}
+
+function AddTodoForm({ addTodo }) {
+  const [description, setDescription] = useState("")
+  const [deadline, setDeadline] = useState("")
+
+  function onSubmit(event) {
+    event.preventDefault()
+
+    addTodo(description, deadline)
+
+    setDescription("")
+    setDeadline("")
+  }
+
+  const editing = event => {
```

Comment:
I'd call this one `onEditDescription` as well

Reply:
Thank you very much for your valuable feedback, I will emplement the changes and merge :)